src/newgrf_config.h

```diff
@@ -242,7 +242,7 @@ std::string GRFBuildParamList(const GRFConfig &c);
 void ShowNewGRFSettings(bool editable, bool show_params, bool exec_changes, GRFConfigList &config);
 void OpenGRFParameterWindow(bool is_baseset, GRFConfig &c, bool editable);
 
-void UpdateNewGRFScanStatus(uint num, const char *name);
+void UpdateNewGRFScanStatus(uint num, std::string name);
```

Comment:
Shouldn't these be `std::string &&name`?

If move semantics would been used in this case, then a function with `std::string &&name` and `std::string name` should be called in the same way, but it doesn't seem to be done: https://godbolt.org/z/Ydj1h4dnb.

Reply:
With a `std::string` parameter the caller can choose to either copy or move the string to the parameter.
With a `std::string&&` there is neither a copy nor a move; the parameter is passed as reference.

Currently the string is moved 3 times:
* In `AddFile` from local `name` to `UpdateNewGRFScanStatus`.
* In `UpdateNewGRFScanStatus` to `ScanProgressWindow::UpdateNewGRFScanStatus`.
* In `ScanProgressWindow::UpdateNewGRFScanStatus` to `last_name`.

If we change the two `UpdateNewGRFScanStatus`, the string will be moved 1 time:
* `AddFile` passes the string as rvalue reference.
* `UpdateNewGRFScanStatus` passes the string as rvalue reference.
* `ScanProgressWindow::UpdateNewGRFScanStatus` moves the string to `last_name`.

Fine by me.